django/db/migrations/state.py

```diff
@@ -444,28 +446,25 @@ def flatten_bases(model):
             bases = (models.Model,)
 
         managers = []
-
-        # Make sure the default manager is always first since ordering chooses
-        # the default manager.
-        if not model._default_manager.auto_created:
-            if model._default_manager.use_in_migrations:
-                default_manager = copy.copy(model._default_manager)
-                default_manager._set_creation_counter()
-
-            # If the default manager doesn't have `use_in_migrations = True`,
-            # shim a default manager so another manager isn't promoted in its
-            # place.
+        default_manager_shim = None
+        for manager in model._meta.managers:
+            if manager.use_in_migrations:
+                new_manager = copy.copy(manager)
+                new_manager._set_creation_counter()
+            elif manager is model._base_manager or manager is model._default_manager:
+                new_manager = models.Manager()
+                new_manager.model = manager.model
+                new_manager.name = manager.name
+                if manager is model._default_manager:
+                    default_manager_shim = new_manager
             else:
-                default_manager = models.Manager()
-                default_manager.model = model
-                default_manager.name = model._default_manager.name
-            managers.append((force_text(default_manager.name), default_manager))
+                continue
+            managers.append((force_text(manager.name), new_manager))
 
-        for manager in model._meta.managers:
-            if manager.use_in_migrations and manager is not model._default_manager:
-                manager = copy.copy(manager)
-                manager._set_creation_counter()
-                managers.append((force_text(manager.name), manager))
+        # If all we have is a shimmed default manager called objects, we can just
```

Comment:
Lately I've noticed that avoiding "we" in comments often results in a simpler wording, e.g. `# Ignore a shimmed default manager called objects if it's the only one.`